tests/unit/test_contracts/explicit_nested_tests.cpp

```diff
@@ -93,6 +93,20 @@ CONTRACT explicit_nested_tests : public contract {
       return output;
    }
 
+   [[eosio::action]]
+   // usage : cleos -v push action eosio vmistr '[[[{"key":12,"value":"map test"},{"key":34,"value":"passed"}]]]'  -p eosio@active
+   std::vector<std::map<int, std::string>>  vmistr(std::vector<std::map<int, std::string>> input){
+      std::vector<std::map<int, std::string>> output = input;
+      return output;
+   }
+
+   [[eosio::action]]
+   // usage : cleos -v push action eosio vpistr '[[{"first":12,"second":"pair test"},{"first":34,"second":"passed"}]]'  -p eosio@active
+   std::vector<std::pair<int, std::string>>  vpistr(std::vector<std::pair<int, std::string>> input){
+      std::vector<std::pair<int, std::string>> output = input;
+      return output;
+   }
+
```

Comment:
I suggest you add at least two more cases here: map of pairs, pair of maps, see https://github.com/EOSIO/eosio.cdt/blob/develop/tests/toolchain/build-pass/nestcontn2a.cpp ans search for set mp and setpm to know their input formats,  it is even better you can try to add map of maps, pair of pairs as shown in setmm and setpp of the same link

Reply:
Good idea.

Reply:
Test cases added, and per my test it passed.